tests/foreman/api/test_registration.py

```diff
@@ -107,12 +107,22 @@ def test_positive_allow_reregistration_when_dmi_uuid_changed(
     """
     uuid_1 = str(uuid.uuid1())
     uuid_2 = str(uuid.uuid4())
-    rhel_contenthost.install_katello_ca(target_sat)
     target_sat.execute(f'echo \'{{"dmi.system.uuid": "{uuid_1}"}}\' > /etc/rhsm/facts/uuid.facts')
-    result = rhel_contenthost.register_contenthost(module_org.label, lce=ENVIRONMENT)
-    assert result.status == 0
-    result = rhel_contenthost.execute('subscription-manager clean')
+    command = target_sat.api.RegistrationCommand(
+        organization=module_org,
+        activation_keys=[module_ak_with_synced_repo.name],
+        location=module_location,
+    ).create()
+    result = rhel_contenthost.execute(command)
     assert result.status == 0
+    rhel_contenthost.unregister()
```

Comment:
Instead of unregister, can't we force register in next iteration?

Reply:
I would advocate against force unregister in any kind of automation. In my view,  `force` options are to be used by people not programs.

Reply:
I agree with @ogajduse. We don't need to force register it. It will be better to unregister it first and then register the host after changing dmi_uuid.

Reply:
@ogajduse force param is part of registration form, so this is intended for testing by programs, and either we use clean or force both do the same thing for host :)
but I agree with @shweta83 if it isn't required here, as we need to register after changeing uuid, so it makes sense.